### DISCUSSION TITLE
Failed to restore a backup

### DISCUSSION THREAD

**Discussion by mlnotes** [teslamate-org/teslamate#2862] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Failed to restore a backup

Data backup from v1.26.1, Restore to v1.27.1

Restore in window dock, commmand
```
 Get-Content .\teslamate.bck | docker-compose exec -T database psql -U teslamate -d teslamate
```

### Expected Behavior

Should restore successfully

### Relevant log output

```shell
Error log:

ALTER TABLE
ERROR:  value too long for type character varying(255)
CONTEXT:  COPY addresses, line 1, column state_district: "{"lat": "31.233656872832896", "lon": "121.5042941190686", "name": "?????, "type": "secondary", "osm_..."
COPY 1
COPY 1
COPY 8947
COPY 40
COPY 225
ERROR:  numeric field overflow
DETAIL:  A field with precision 8, scale 6 must round to an absolute value less than 10^2.
CONTEXT:  COPY geofences, line 5, column latitude: "121.241356"
COPY 866077
COPY 85
COPY 1
COPY 420
COPY 1
COPY 1
 setval
--------
     87
(1 row)
```

```
ERROR:  insert or update on table "charging_processes" violates foreign key constraint "charging_processes_address_id_fkey"
DETAIL:  Key (address_id)=(27) is not present in table "addresses".
ALTER TABLE
ERROR:  insert or update on table "charging_processes" violates foreign key constraint "charging_processes_geofence_id_fkey"
DETAIL:  Key (geofence_id)=(2) is not present in table "geofences".
ALTER TABLE
ALTER TABLE
ERROR:  insert or update on table "drives" violates foreign key constraint "drives_end_address_id_fkey"
DETAIL:  Key (end_address_id)=(2) is not present in table "addresses".
ERROR:  insert or update on table "drives" violates foreign key constraint "drives_end_geofence_id_fkey"
DETAIL:  Key (end_geofence_id)=(3) is not present in table "geofences".
ALTER TABLE
ERROR:  insert or update on table "drives" violates foreign key constraint "drives_start_address_id_fkey"
DETAIL:  Key (start_address_id)=(1) is not present in table "addresses".
ERROR:  insert or update on table "drives" violates foreign key constraint "drives_start_geofence_id_fkey"
DETAIL:  Key (start_geofence_id)=(3) is not present in table "geofences".
```
```


### Screenshots

_No response_

### Additional data

_No response_

### Type of installation

Docker

### Version

v1.27.1

**Comment by ktsenvolt** [comment]
# Stop the teslamate container to avoid write conflicts
docker-compose stop teslamate

# Drop existing data and reinitialize
docker-compose exec -T database psql -U teslamate << .
drop schema public cascade;
create schema public;
create extension cube;
create extension earthdistance;
CREATE OR REPLACE FUNCTION public.ll_to_earth(float8, float8)
    RETURNS public.earth
    LANGUAGE SQL
    IMMUTABLE STRICT
    PARALLEL SAFE
    AS 'SELECT public.cube(public.cube(public.cube(public.earth()*cos(radians(\$1))*cos(radians(\$2))),public.earth()*cos(radians(\$1))*sin(radians(\$2))),public.earth()*sin(radians(\$1)))::public.earth';
.

# Restore
docker-compose exec -T database psql -U teslamate -d teslamate < teslamate.bck

# Restart the teslamate container
docker-compose start teslamate

**Comment by KelaKing** [comment]
I have the same issue, I can restore the data, but lose some column
For example the Start and Destination

![image](https://user-images.githubusercontent.com/13584044/232688110-591afc9e-20e8-48d0-a9e0-a9bbb1b09e60.png)

**Comment by pencilheart** [comment]
Also miss location
